### PR TITLE
Feature/private event code review

### PR DESCRIPTION
**Pull request by majovainilla**
Requesting code review for Event App that includes:

- [x] Setup and Sign In
      - Using cookies and sesssions
      - Includes Sign In and Log In links

- [x] Basic Events
    - Create event with associated user_id (creator)

- [x] Event Attendance
   - Use of associations with 'through table'
   -  Join event

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class EventsController < ApplicationController
+  def new
+    @event = Event.new
```
[the remark this answers is not in the document]
Comment (majovainilla) [reply app/controllers/events_controller.rb]: Thanks for the suggestion! It's done =)

File: app/models/event.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+class Event < ApplicationRecord
+  belongs_to :creator, class_name: 'User'
+  has_many :attendances, foreign_key: :attended_event
+  has_many :attendees, through: :attendances, source: 'attendee'
+
+  validates :name, :description, presence: true
+  scope :previous, -> { where('date < ?', Date.today).order(date: :asc) }
+  scope :upcoming, -> { where('date > ?', Date.today).order(date: :asc) }
+
+  # def self.previous_events
```
[the remark this answers is not in the document]
Comment (majovainilla) [reply app/models/event.rb]: Done!

**Comment by majovainilla** [comment]
HELP WITH TEST:

Dear TSE, we would like your assistance with this failing test. We did our research and none of the suggestions worked.  

```
 Failure/Error: it { should validate_uniqueness_of(:attendee_id).scoped_to(:attended_event_id) }
     
     ActiveRecord::StatementInvalid:
       SQLite3::SQLException: no such column: attendances.
```

THanks!

**Comment by samgaco** [comment]
Hi @majovainilla and @fabricio-garcia :high_brightness: 

All the changes were implemented correctly so I will let it approved. And when it comes to that test, please use the [forum](https://questions.microverse.org/login) for technical questions or [book time](https://microverse.zendesk.com/hc/en-us/articles/360026625973-How-can-I-get-help-from-TSEs-) with a full-time TSE to see how to go about with the issue.

---------------------------

* ###### STATUS OF THE PROJECT: APPROVED :heavy_check_mark:

**Pull request merged**